### ISSUE TITLE
Generalize CSP model

### ISSUE THREAD

**Issue by davide-f** [pz-max/pypsa-africa#1038]
## Describe the feature you'd like to see

Extend the csp modelling to support:
- multiple CSP technologies at the same time
- customizability of tower/storage/generation unit

To tackled this, the code shall be revised to support "csp-{technology}" options

**Comment by davide-f** [comment]
FYI @pz-max @ekatef @GbotemiB

**Comment by ekatef** [comment]
@davide-f thanks a lot for the issue 🙂

Do you think it makes sense add also the points to consider more advanced modelling features, as we discussed:
- Land-use availability for CSP should include the slope, which can be implemented [using gebco dataset](https://github.com/euronion/trace/blob/1b313271081a04e1fef6090ef3ec494489021035/config/config.default.yaml#L235-L240).
- Some additionally constraints should be applied for CSP model to make sure that its operation is more realistic from thermal engineering perspective.

**Comment by davide-f** [comment]
Good point!
The slope issue I think deserve an issue per se, as it applies also to other technologies beyond csp.
For the second point, as well I think it can be a different issue and treated separately.
What do you think?

**Comment by ekatef** [comment]
> * Land-use availability for CSP should include the slope, which can be implemented [using gebco dataset](https://github.com/euronion/trace/blob/1b313271081a04e1fef6090ef3ec494489021035/config/config.default.yaml#L235-L240).

True, both points have appeared in a discussion around CSP but are actually of different nature. Added as #1039 and #1040 🙂
